hr_expense_advance_clearing/models/hr_expense.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Kitti Upariphutthiphong <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    advance = fields.Boolean(
+        string='Employee Advance',
+        default=False,
+    )
+
+    @api.multi
+    @api.constrains('advance')
+    def _check_advance(self):
+        for expense in self:
```

Comment:
```suggestion
        for expense in self.filtered('advance'):
```

To avoid the next `if`